aiida/backends/sqlalchemy/models/comment.py

```diff
@@ -28,18 +29,30 @@ class DbComment(Base):
 
     id = Column(Integer, primary_key=True)  # pylint: disable=invalid-name
 
-    uuid = Column(UUID(as_uuid=True), default=get_new_uuid, unique=True)
-    dbnode_id = Column(Integer, ForeignKey('db_dbnode.id', ondelete='CASCADE', deferrable=True, initially='DEFERRED'))
+    uuid = Column(UUID(as_uuid=True), default=get_new_uuid, nullable=False)
+    dbnode_id = Column(
+        Integer, ForeignKey('db_dbnode.id', ondelete='CASCADE', deferrable=True, initially='DEFERRED'), nullable=False
+    )
 
-    ctime = Column(DateTime(timezone=True), default=timezone.now)
-    mtime = Column(DateTime(timezone=True), default=timezone.now, onupdate=timezone.now)
+    ctime = Column(DateTime(timezone=True), default=timezone.now, nullable=False)
+    mtime = Column(DateTime(timezone=True), default=timezone.now, onupdate=timezone.now, nullable=False)
 
-    user_id = Column(Integer, ForeignKey('db_dbuser.id', ondelete='CASCADE', deferrable=True, initially='DEFERRED'))
-    content = Column(Text, nullable=True)
+    user_id = Column(
+        Integer, ForeignKey('db_dbuser.id', ondelete='CASCADE', deferrable=True, initially='DEFERRED'), nullable=False
+    )
+    content = Column(Text, default='', nullable=False)
 
     dbnode = relationship('DbNode', backref='dbcomments')
     user = relationship('DbUser')
 
+    __table_args__ = (
+        # index/constraint names mirror django's auto-generated ones
+        UniqueConstraint(uuid, name='db_dbcomment_uuid_49bac08c_uniq'),
```

Comment:
If we are going to keep the SqlAlchemy implementation, why are we migrating the SqlAlchemy databases to match the Django one? Wouldn't it be way more logical to migrate Django to match SqlAlchemy?

Reply:
Because you cannot control the name of indexes/constraints on the django ORM; they are auto-generated in private methods

Reply:
Note, you could change the names of indexes/constraints in django migrations, but then the schema you create via the migration route would be different to the one you create via the models.
Although this parity is not actually tested on the django side, as opposed to on the sqlalchemy side it is specifically tested in `tests/backends/aiida_sqlalchemy/migrations/test_schemas.py::TestMigrationSchemaVsModelsSchema::test_model_and_migration_schemas_are_the_same`

Reply:
(As noted in https://github.com/aiidateam/aiida-core/pull/5097#issuecomment-993200033, this parity is actually already violated in a number of django migrations, so perhaps there should be an additional django migration to fix this 🤷)

Reply:
Ok, regarding the last comment, I think you have now fixed this, correct? That's the change in the `0048_computer_name_to_label.py ` migration?

Once we have dropped Django, would it make sense to revert some of these changes and no longer define the name explicitly? It really feels counterintuitive to have manually define an identifier with a random string.

Reply:
> It really feels counterintuitive to have manually define an identifier with a random string.

I would note here that these strings are not random, they are deterministically generated from the table name, column name(s) and suffix (see https://github.com/django/django/blob/43289707809c814a70f0db38ca4f82f35f43dbfd/django/db/backends/base/schema.py#L989)

If there was a django migration, to fix naming from `aiida/backends/djsite/db/migrations/0021_dbgroup_name_to_label_type_to_type_string.py` and `aiida/backends/djsite/db/migrations/0030_dbnode_type_to_dbnode_node_type.py`,
then it may be possible, with https://docs.sqlalchemy.org/en/14/core/constraints.html#creating-custom-tokens-for-naming-conventions, to actually setup up sqlalchemy to auto-generate exactly the same names.

Although, yeh don't know if it is worth the effort